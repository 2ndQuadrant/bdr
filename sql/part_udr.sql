\c postgres
SELECT bdr.bdr_unsubscribe('node-pg');

-- wait till all slots are killed, we need a better way for that.
SELECT pg_sleep(1);

-- the node state for the removed node should show 'k'
SELECT node_name, node_status FROM bdr.bdr_nodes;

\c regression

-- There should now be zero slots
SELECT * FROM pg_replication_slots;
-- Zero active connections
SELECT count(*) FROM pg_stat_replication;

SELECT node_name, node_status FROM bdr.bdr_nodes;
