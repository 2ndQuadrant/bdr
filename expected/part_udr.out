\c postgres
SELECT bdr.bdr_unsubscribe('node-pg');
 bdr_unsubscribe 
-----------------
 
(1 row)

-- wait till all slots are killed, we need a better way for that.
SELECT pg_sleep(1);
 pg_sleep 
----------
 
(1 row)

-- the node state for the removed node should show 'k'
SELECT node_name, node_status FROM bdr.bdr_nodes;
     node_name      | node_status 
--------------------+-------------
 node-pg            | k
 node-pg-subscriber | k
(2 rows)

\c regression
-- There should now be zero slots
SELECT * FROM pg_replication_slots;
 slot_name | plugin | slot_type | datoid | database | active | xmin | catalog_xmin | restart_lsn 
-----------+--------+-----------+--------+----------+--------+------+--------------+-------------
(0 rows)

-- Zero active connections
SELECT count(*) FROM pg_stat_replication;
 count 
-------
     0
(1 row)

SELECT node_name, node_status FROM bdr.bdr_nodes;
 node_name | node_status 
-----------+-------------
(0 rows)

